ingest_alos2_proto.py

```diff
@@ -0,0 +1,300 @@
+# TAR_TYPE = ["tbz2", "tgz", "bz2", "gz"]
+# ALL_TYPES.extend(TAR_TYPE)
+
+
+def verify(path, file_type):
+    """Verify downloaded file is okay by checking that it can
+       be unzipped/untarred."""
+
+    test_dir = "./extract_test"
+    if file_type in ZIP_TYPE:
+        if not zipfile.is_zipfile(path):
+            raise RuntimeError("%s is not a zipfile." % path)
+        with zipfile.ZipFile(path, 'r') as f:
+            f.extractall(test_dir)
+        shutil.rmtree(test_dir, ignore_errors=True)
+    else:
+        raise NotImplementedError("Failed to verify %s is file type %s." % \
+                                  (path, file_type))
+
+
+def extract(zip_file):
+    """Extract the zipfile."""
+
+    with zipfile.ZipFile(zip_file, 'r') as zf:
+        prod_dir = zip_file.replace(".zip", "")
+        zf.extractall(prod_dir)
+    return prod_dir
+
+
+def download(download_url, dest, oauth_url):
+    # download
+    logging.info("Downloading %s to %s." % (download_url, dest))
+    try:
+        osaka.main.get(download_url, dest, params={"oauth": oauth_url}, measure=True, output="./pge_metrics.json")
+    except Exception, e:
+        tb = traceback.format_exc()
+        logging.error("Failed to download %s to %s: %s" % (download_url,
+                                                           dest, tb))
+        raise
+
+
+def create_metadata(alos2_md_file, download_url):
+    # TODO: Some of these are hardcoded! Do we need them?
+    metadata = {}
+    # open summary.txt to extract metadata
+    # extract information from summary see: https://www.eorc.jaxa.jp/ALOS-2/en/doc/fdata/PALSAR-2_xx_Format_GeoTIFF_E_r.pdf
+    logging.info("Extracting metadata from %s" % alos2_md_file)
+    dummy_section = "summary"
+    with open(alos2_md_file, 'r') as f:
+        # need to add dummy section for config parse to read .properties file
+        summary_string = '[%s]\n' % dummy_section + f.read()
+    summary_string = summary_string.replace('"', '')
+    buf = StringIO.StringIO(summary_string)
+    config = ConfigParser.ConfigParser()
+    config.readfp(buf)
+
+    # parse the metadata from summary.txt
+    alos2md = {}
+    for name, value in config.items(dummy_section):
+        alos2md[name] = value
+
+    metadata['alos2md'] = alos2md
+
+    # facetview filters
+    dataset_name = metadata['alos2md']['scs_sceneid'] + "_" + metadata['alos2md']['pds_productid']
+    metadata['prod_name'] = dataset_name
+    metadata['spacecraftName'] = dataset_name[0:5]
+    metadata['dataset_type'] = dataset_name[0:5]
+    metadata['orbitNumber'] = int(dataset_name[5:10])
+    metadata['scene_frame_number'] = int(dataset_name[10:14])
+    prod_datetime = datetime.datetime.strptime(dataset_name[15:21], '%y%m%d')
+    prod_date = prod_datetime.strftime("%Y-%m-%d")
+    metadata['prod_date'] = prod_date
+
+    # TODO: not sure if this is the right way to expose this in Facet Filters, using CSK's metadata structure
+    dfdn = {"AcquistionMode": dataset_name[22:25],
```

Comment:
Not sure if this is the right way to expose this field in FacetView's Filters, using CSK's metadata structure. ALOS2 has several acq modes which should be displayed on the LHS filter panels. Do we change tosca's code (index.html?)